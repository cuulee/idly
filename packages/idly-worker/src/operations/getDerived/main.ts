import { parseResponse } from '../../misc/channelBuilder';
import { MainOperation } from '../helpers';
import { OperationKinds } from '../types';
import { GetDerived } from './type';

export const getDerived: (con: any) => MainOperation<GetDerived> = (
  connector: any
) => parseResponse(connector, OperationKinds.GetDerived);
