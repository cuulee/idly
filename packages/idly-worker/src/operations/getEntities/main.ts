import { Shrub } from 'idly-common/lib/state/graph/shrub';

import { getChannelBuilder } from '../../misc/channelBuilder';
import { GetActions, Operation } from '../operationsTypes';
import { GetEntities } from './type';

export function getEntities(connector: any): Operation<GetEntities> {
  const channel = getChannelBuilder<GetEntities>(connector)(
    GetActions.GetEntities,
  );
  return async request => {
    const json = await channel(request);
    const parsedTree: GetEntities['response'] = Shrub.fromString(json);
    return parsedTree;
  };
}
