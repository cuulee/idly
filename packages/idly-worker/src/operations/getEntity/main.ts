
import { parseResponse } from '../../misc/channelBuilder';
import { MainOperation } from '../helpers';
import { OperationKinds } from '../types';
import { GetEntity } from './type';

export const getEntity: (con: any) => MainOperation< GetEntity> = (
    connector: any
) => parseResponse(connector, OperationKinds.GetEntity);
