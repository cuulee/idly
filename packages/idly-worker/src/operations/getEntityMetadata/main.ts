import { parseResponse } from '../../misc/channelBuilder';
import { MainOperation } from '../helpers';
import { OperationKinds } from '../types';
import { GetEntityMetadata } from './type';

export const getEntityMetadata: (
  con: any
) => MainOperation<GetEntityMetadata> = (connector: any) =>
  parseResponse(connector, OperationKinds.GetEntityMetadata);
