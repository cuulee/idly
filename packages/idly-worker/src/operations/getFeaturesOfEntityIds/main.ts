import { getChannelBuilder } from '../../misc/channelBuilder';
import { GetActions, Operation } from '../operationsTypes';
import { GetFeaturesOfEntityIds } from './type';

export function getFeaturesOfEntityIds(
  connector: any,
): Operation<GetFeaturesOfEntityIds> {
  const channel = getChannelBuilder<GetFeaturesOfEntityIds>(connector)(
    GetActions.GetFeaturesOfEntityIds,
  );
  return async request => {
    const json = await channel(request);
    const parsedFeatures: GetFeaturesOfEntityIds['response'] = JSON.parse(json);
    return parsedFeatures;
  };
}
