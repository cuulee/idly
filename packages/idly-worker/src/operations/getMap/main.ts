import { getChannelBuilder } from '../../misc/channelBuilder';
import { GetActions, Operation } from '../operationsTypes';
import { GetMap } from './type';

/** Main Thread */
export function getMap(connector: any): Operation<GetMap> {
  const channel = getChannelBuilder<GetMap>(connector)(GetActions.GetMap);
  return async req => {
    const json = await channel(req);
    const fc: GetMap['response'] = JSON.parse(json);
    return fc;
  };
}
