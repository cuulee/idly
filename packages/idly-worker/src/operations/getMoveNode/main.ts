import { getChannelBuilder } from '../../misc/channelBuilder';
import { GetActions, Operation } from '../operationsTypes';
import { GetMoveNode } from './type';

/** Main Thread */
export function getMoveNode(connector: any): Operation<GetMoveNode> {
  const channel = getChannelBuilder<GetMoveNode>(connector)(
    GetActions.GetMoveNode
  );
  return async req => {
    const json = await channel(req);
    const fc: GetMoveNode['response'] = JSON.parse(json);
    return fc;
  };
}
