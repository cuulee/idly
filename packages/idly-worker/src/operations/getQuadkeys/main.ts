import { parseResponse } from '../../misc/channelBuilder';
import { MainOperation } from '../helpers';
import { OperationKinds } from '../types';
import { GetQuadkey } from './type';

export const getQuadkey: (con: any) => MainOperation<GetQuadkey> = (
  connector: any
) => parseResponse(connector, OperationKinds.GetQuadkey);
