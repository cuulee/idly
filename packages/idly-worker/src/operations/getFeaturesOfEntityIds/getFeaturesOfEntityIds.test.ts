import { XML3 } from '../../misc/fixtures';
import { stubWorkerLogic } from '../../mocks/PromiseWorkerStub';
import { xmlFetchMock } from '../../mocks/xmlFetchMock';
import { setOsmTiles } from '../setOsmTiles/main';

import { getFeaturesOfEntityIds } from './main';

declare var global: any;

describe('getFeaturesOfEntityIds', () => {
  global.fetch = xmlFetchMock(XML3);
  const promiseWorker = stubWorkerLogic();
  test('small xml test', async () => {
    await setOsmTiles(promiseWorker)({
      bbox: [
        -73.98630242339283,
        40.73537277780156,
        -73.98264244865518,
        40.73941515574535,
      ],
      zoom: 17.54,
    });
    const resp1 = getFeaturesOfEntityIds(promiseWorker)({ entityIds: ['n1'] });
    const resp2 = getFeaturesOfEntityIds(promiseWorker)({
      entityIds: ['n2', 'w1'],
    });
    expect(await resp1).toMatchSnapshot();
    expect(await resp2).toMatchSnapshot();
  });
  test.skip('non existent id should not fail', async () => {
    await setOsmTiles(promiseWorker)({
      bbox: [
        -73.98630242339283,
        40.73537277780156,
        -73.98264244865518,
        40.73941515574535,
      ],
      zoom: 17.54,
    });
    const resp1 = getFeaturesOfEntityIds(promiseWorker)({
      entityIds: ['non1'],
    });
    expect(await resp1).toMatchSnapshot();
  });
});
