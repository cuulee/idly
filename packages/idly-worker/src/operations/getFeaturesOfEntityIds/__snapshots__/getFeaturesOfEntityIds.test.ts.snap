// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFeaturesOfEntityIds small xml test 1`] = `
Array [
  Object {
    "geometry": Object {
      "coordinates": Array [
        74.1816382,
        34.530396,
      ],
      "type": "Point",
    },
    "id": "n1",
    "properties": Object {
      "id": "n1",
      "osm_basic--geometry": "point",
      "osm_basic--icon": "circle",
    },
    "type": "Feature",
  },
]
`;

exports[`getFeaturesOfEntityIds small xml test 2`] = `
Array [
  Object {
    "geometry": Object {
      "coordinates": Array [
        74.1816382,
        34.530396,
      ],
      "type": "Point",
    },
    "id": "n2",
    "properties": Object {
      "id": "n2",
      "osm_basic--geometry": "vertex",
      "osm_basic--icon": "circle",
    },
    "type": "Feature",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        74.1816382,
        34.530396,
      ],
      "type": "Point",
    },
    "id": "n1",
    "properties": Object {
      "id": "n1",
      "osm_basic--geometry": "vertex",
      "osm_basic--icon": "circle",
    },
    "type": "Feature",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        Array [
          74.1816382,
          34.530396,
        ],
        Array [
          74.1816382,
          34.530396,
        ],
      ],
      "type": "LineString",
    },
    "id": "w1",
    "properties": Object {
      "id": "w1",
      "osm_basic--geometry": "line",
      "osm_basic--tagsClass": "",
    },
    "type": "Feature",
  },
]
`;
